packages/icon/test/icon.test.js

```diff
@@ -161,6 +166,31 @@ describe('vaadin-icon', () => {
       icon.__fetch.restore();
     });
 
+    it('should append value from symbol property to src', () => {
+      icon.src = './icon.svg';
+      icon.symbol = 'symbol-id';
+
+      expect(svgElement.querySelector(`use[href="${icon.src}#${icon.symbol}"]`)).to.exist;
+    });
+
+    it('should use value from symbol when src path has a hash value', () => {
+      icon.src = './icon.svg#id-0';
+      icon.symbol = 'id-1';
+
+      expect(svgElement.querySelector(`use[href="${icon.src.split('#')[0]}#${icon.symbol}"]`));
+    });
+
+    it('should render SVG content and <use> if src is given in data format with symbol prop defined', async () => {
```

Comment:
Is this use case (or data urls for src in general) something we want to support? Asking since it seems to require making the icon internals more complex.

I'd imagine SVG sprite as a data url + symbol id is a combination one wouldn't use because it doesn't win you anything. If you have multiple icons, you'd still need to pass the data url to each icon separately.

Reply:
I mentioned before in the daily and some of the meetings, that as an example in the docs project when you import an SVG file (`import file from 'path/to/file.svg'`), the value returned is in `data:image...` format, instead of just the path to the file. If the file is a sprite, there's no way to define the symbol id if we don't support this case.

Reply:
Ideally, the import would return the path instead, as it does with `png` files in the avatar docs, for instance. Then we wouldn't need to support this case.

Reply:
One option could be to use an SVG from a CDN in the docs, like https://unpkg.com/@fortawesome/fontawesome-free@6.4.2/sprites/solid.svg We haven't generally used external resources in the docs though

Reply:
It's not allowed to use `<use>` with cross-origin URL.

Reply:
Right, it's not an option then either. One way could be to serve the SVG as a static file from the Vaadin project (`docs`). But the current approach is probably the least hacky for now.

I guess it's up to the bundler/plugins how an import of an SVG file is handled. In this case, it resolves as a data URL whereas an import of a PNG file resolves as a path to a generated static file.

Reply:
Oh, I just realized that with this new implementation, we actually can support external SVG sprites. See the dev page of the prototype branch at https://github.com/vaadin/web-components/tree/experiment/icon/external-svg-sprite

Reply:
Nice. One benefit of using the browser is that we can rely on its implementation of how it handles the fetching in case of multiple instances of `<vaadin-icon>` using the same SVG file. Doing this way, we would be probably spanning multiple requests at the same time, and it would save the whole sprite content in each instance. Maybe this could be an option under a new `cross-origin` property?

Reply:
If we wanted to make it work efficiently, we could internally cache the requests/results to a url and not execute a fetch if the same svg has already been requested/fetched.

This would also benefit the current implementation in scenarios where you use the same (non sprite) SVG on a page more than once.

Reply:
Yeah, I also considered caching it internally.